programs/summerofcode/2025.md

```diff
@@ -177,6 +177,37 @@ Note that the initial idea is to solve this with **3-way Git merges**. However,
 
 - Upstream Issue: [WIP - Proposal: Automating Operator Maintenance: Driving Better Results with Less Overhead](https://github.com/kubernetes-sigs/kubebuilder/pull/4302)
 
+#### KubeStellar
+
+##### AI/ML Model Monitoring and Drift Detection in Disconnected Clusters using KubeStellar
+
+- Description: AI/ML models deployed in disconnected environments, such as edge clusters and air-gapped systems, often suffer from model drift—a degradation in model performance due to changes in input data distributions. Without continuous monitoring, models may become inaccurate, leading to unreliable predictions.
+
+This project aims to integrate model monitoring and drift detection into KubeStellar, enabling Kubernetes-based AI workloads to detect data drift locally and sync monitoring metrics when connectivity is restored. The solution will use lightweight monitoring agents deployed alongside ML models to track data distribution changes and alert mechanisms to trigger model retraining when necessary.
+
+The system will also include policies for efficient metric storage and synchronization between disconnected and central clusters while minimizing bandwidth usage.
+
+- Expected Outcome:
+ - A KubeStellar-compatible AI/ML monitoring component that tracks model drift in disconnected clusters.
+ - Efficient local storage and synchronization of monitoring metrics when connectivity is restored.
+ - Policies for adaptive model retraining triggers based on drift detection signals.
+ - Integration with existing ML tools (e.g., Prometheus, TensorFlow Extended, OpenTelemetry).
+ - Open-source documentation and example workflows demonstrating how KubeStellar manages AI model monitoring across disconnected clusters.
+- Recommended Skills:
+ - Kubernetes and container orchestration
+ - AI/ML model deployment & monitoring
+ - Python, Go (for Kubernetes integrations)
+ - Experience with logging/monitoring tools (Prometheus, OpenTelemetry)
+ - Familiarity with KubeStellar (preferred but not required)
+- Expected Project Size: Large (~350 hours)
+This project requires implementing multiple components: local monitoring, drift detection, synchronization, and integration with KubeStellar. It also involves research into efficient data synchronization strategies for low-bandwidth environments.
+
+- Mentor(s):
+Andy Anderson (@clubanderson, andy@clubanderson.com) - Primary Mentor
+[Second Mentor's Name] (@second-mentor-github, second-mentor-email)
```

Comment:
We can merge this in without the second mentor for now, but for GSoC it's important for us to have one.

Reply:
understood